hpx/util/coroutine/detail/context_generic_context.hpp

```diff
@@ -232,8 +233,7 @@ namespace hpx { namespace util { namespace coroutines
             std::ptrdiff_t get_available_stack_space()
             {
 #if defined(HPX_HAVE_THREADS_GET_STACK_POINTER)
-                return
-                    reinterpret_cast<uintptr_type>(stack_pointer_) - get_stack_ptr();
+                return get_stack_ptr() - reinterpret_cast<std::size_t>(stack_pointer_);
```

Comment:
Yes, this was on purpose since the stack_pointer_ here points to the top of stack already, while the other ones point to the bottom.